scan.py

```diff
@@ -32,6 +33,25 @@ def event_object(event):
         raise Exception("Unable to retrieve object from event.")
     return s3.Object(bucket, key)
 
+def verify_s3_object_version(s3_object):
+    # validate that we only process the original version of a file, if asked to do so
+    # security check to disallow processing of a new (possibly infected) object version
+    # while a clean initial version is getting processed
+    # downstream services may consume latest version by mistake and get the infected version instead
+    if str_to_bool(AV_PROCESS_ORIGINAL_VERSION_ONLY):
+        bucketVersioning = s3.BucketVersioning(s3_object.bucket_name)
+        if (bucketVersioning.status == "Enabled"):
+            bucket = s3.Bucket(s3_object.bucket_name)
+            versions = list(bucket.object_versions.filter(Prefix=s3_object.key))
+            if len(versions) > 1:
+                print("Detected multiple object versions in %s.%s, aborting processing" % (s3_object.bucket_name, s3_object.key))
+                raise Exception("Detected multiple object versions in %s.%s, aborting processing" % (s3_object.bucket_name, s3_object.key))
+            else:
+                print("Detected only 1 object version in %s.%s, proceeding with processing" % (s3_object.bucket_name, s3_object.key))
```

Comment:
could suppress this as it will unnecessary add to log message which couldn't add much value.

Reply:
technically this should never happen, as UI should ask for new upload slot every time. If a second version is getting uploaded, most likely someone is trying to hack it directly via the UI.

So this should be a very unusual occurrence and hence better to log it.